docs/testing/runtimes.md

```diff
@@ -0,0 +1,117 @@
+- To share data between a task pane and a function command, store data in [Window.localStorage](https://developer.mozilla.org/docs/Web/API/Window/localStorage), which is shared across all runtimes that access the same specific [origin](https://developer.mozilla.org/docs/Glossary/Origin). *LocalStorage isn't accessible in a JavaScript-only runtime and, thus, it isn't available in Excel custom functions or Outlook event-based tasks.*
+
+    > [!NOTE]
+    > Data in `Window.localStorage` persists between sessions of the add-in and is shared by add-ins with the same origin. Both of these characteristics are often undesirable for an add-in. You can ensure that each session of a given add-in starts fresh by calling the [Window.localStorage.clear](https://developer.mozilla.org/docs/Web/API/Storage/clear) method when the add-in starts. To allow some stored values to persist, but reinitialize other values, you can use [Window.localStorage.setItem](https://developer.mozilla.org/docs/Web/API/Storage/setItem) when the add-in starts for each item that should be reset to an initial value. You can also call [Window.localStorage.removeItem](https://developer.mozilla.org/docs/Web/API/Storage/removeItem) to delete an item entirely.
+
+- To share data between an Excel custom function and any other runtime, use [OfficeRuntime.storage](/javascript/api/office-runtime/officeruntime.storage).
+- To share data between an Outlook event-based task and any other runtime, use [OfficeRuntime.storage](/javascript/api/office-runtime/officeruntime.storage).
+
+    > [!NOTE]
+    > The `OfficeRuntime.storage` methods cannot be called in a dialog, so this is not an option for sharing data between a dialog and another runtime. 
+
+Other ways to share data include the following:
+
+- Store shared data in an online database that is accessible to all the runtimes.
+- Store shared data in a cookie for the add-in's domain.
+
+For more information, see [Persist add-in state and settings](../develop/persisting-add-in-state-and-settings.md) and [Manage state and settings for an Outlook add-in](../outlook/manage-state-and-settings-outlook.md).
+
+## JavaScript-only runtime
+
+The JavaScript-only runtime that is used in Office add-ins is a modification of an open source runtime originally created for [React Native](https://reactnative.dev/). It contains a JavaScript engine supplemented with support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage). It doesn't have a rendering engine. 
+
+This type of runtime is used in Outlook event-based (autolaunch) tasks and in Excel custom functions *except* when the custom function is [sharing a runtime](#shared-runtime). 
```

Comment:
Should we specify Windows here since the runtimes are different when running Office for Mac?